consensus/wal.go

```diff
@@ -166,17 +167,25 @@ func (wal *baseWAL) SearchForEndHeight(height int64, options *WALSearchOptions)
 		for {
 			msg, err = dec.Decode()
 			if err == io.EOF {
+				// OPTIMISATION: no need to look for height in older files if we've seen h < height
+				if lastHeightFound > 0 && lastHeightFound < height {
+					gr.Close()
+					return nil, false, nil
+				}
 				// check next file
 				break
 			}
 			if options.IgnoreDataCorruptionErrors && IsDataCorruptionError(err) {
+				wal.Logger.Debug("Corrupted entry. Skipping...", "err", err)
 				// do nothing
+				continue
 			} else if err != nil {
 				gr.Close()
 				return nil, false, err
 			}
 
 			if m, ok := msg.Msg.(EndHeightMessage); ok {
+				lastHeightFound = m.Height
 				if m.Height == height { // found
```

Comment:
better yet why not just exit on the else condition here ?

Reply:
hm.. but we need to read until the end to ensure there is no `height`

Reply:
if we just check else if `m.Height < height` and return, we might return false when if fact height exists...

Reply:
but we start from the top of the file right ? and the WAL already guarantees the height is increasing. so if we find a height, and its less than what we're looking for, we should be done. no ?

Reply:
1 -> 2 -> 3

we're looking for 3. if we return on 1 or 2, it would be incorrect since there is 3

Reply:
yeh duh. larger heights are at the bottom of the file.